web/src/test/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilterTests.java

```diff
@@ -217,5 +319,28 @@ private FilterAndTelemetryClientMock createInitializedFilterWithMockTelemetryCli
 
         return new FilterAndTelemetryClientMock(filter, mockTelemetryClient);
     }
+
+
+    /*For Parametrized constructor testing*/
+    private FilterAndTelemetryClientMock createInitializedParametrizedFilterWithMockTelemetryClient(boolean withTelemetryClient, boolean clientThrows) throws ServletException, NoSuchFieldException, IllegalAccessException {
+        Filter filter = createParametrizedInitializedFilter("My-App");
```

Comment:
There is no validation that "My-app" has actually made it though and changed the way the code works.
You may want to add a test that checks my-app assignment or tests a change in the execution flow caused by the name being pre-assigned.